src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SearchRequestProcessor.java

```diff
@@ -205,6 +206,9 @@ private static Map<String, FieldDef> getVirtualFields(
   private static Map<String, FieldDef> getRetrieveFields(
       SearchRequest request, Map<String, FieldDef> queryFields) {
     Map<String, FieldDef> retrieveFields = new HashMap<>();
+    if (request.getRetrieveFieldsList().equals(Arrays.asList("*"))) {
```

Comment:
This would create a new single element list each time this code gets executed. You can instead check the first element of the list without any new object creation.
Also you can declare the WILDCARD string as a static variable and use it here.
`request.getRetrieveFieldsCount() == 1 && request.getRetrieveFields(0).equals(WILDCARD)`

Reply:
This looks more efficient, thanks!